test/test_core.py

```diff
@@ -5873,8 +5877,12 @@ def test_fs_mmap(self):
     self.uses_es6 = True
     for fs in ['MEMFS', 'NODEFS', 'NODERAWFS']:
       if fs == 'NODEFS':
+        if not self.is_node():
```

Comment:
This checks if there is 1 JS engine and it is node, but what if the list of engines is `[node, other]`?

Reply:
That would fail since the non-node engine could not run the code.   This part of the test requires that node (and only node) is used to run the test.

Running with more than one engine configured is very rare.. we have no testing for that configuration that I know of.  Do you often do it?  What would you expect to happen in that case?   Perhaps these tests should narrow the list, like `require_node` does?

Reply:
I think we have `--all-engines` for multi-engine testing? I don't do it often since engine bugs are so rare, though.

Narrowing the list seems better to me, good idea.

Another option might be to add code in a single place that does that narrowing automatically if `NODEFS` is in the commandline arguments?

Reply:
> I think we have `--all-engines` for multi-engine testing? I don't do it often since engine bugs are so rare, though.
> 

The problem with `--all-engines` is that we don't test it, so at any given time is likely broken.

I wonder if its worth removing it for this reason?  What are the chances that the entire `other` or `core` test suite will run will `--all-engines` set?

Having said that i think can do the narrowing thing here..

Reply:
> What are the chances that the entire other or core test suite will run will --all-engines set?

Maybe the chances aren't high, since we don't test it. I guess it's probably fine at this point to remove all-engines - we're committed to running most of the tests in node, and only a select few in v8.

Reply:
I remove this part of the change.